apps/microtvm/arduino/template_project/microtvm_api_server.py

```diff
@@ -177,11 +177,28 @@ def _remove_unused_components(self, source_dir, project_type):
         for component in unused_components:
             shutil.rmtree(source_dir / "standalone_crt" / component)
 
+    def _safe_extract(tar, path=".", members=None, *, numeric_owner=False):
+        def is_within_directory(directory, target):
+
+            abs_directory = os.path.abspath(directory)
+            abs_target = os.path.abspath(target)
+
+            prefix = os.path.commonprefix([abs_directory, abs_target])
+
+            return prefix == abs_directory
+
+        for member in tar.getmembers():
+            member_path = os.path.join(path, member.name)
```

Comment:
Could we move this path join to `is_within_directory()` to make it just a bit more self-contained?